### PR TITLE
Feature/job name

### LINKED ISSUE #484: feature request: job_name attribute in resource description
See radical-collaboration/hpc-workflows/issues/122: EnTK should forward `job_name` settings to RP.

### PR DESCRIPTION
**Pull request by iparask**
This PR introduces the `job_name` attribute when a resource dict is defined.

It closes #484.

### CODE REVIEW

**Comment by andre-merzky** [comment]
@iparask , @lee212 : please make sure this goes into the next release.  I *think* it should be ready to be merged?

**Comment by iparask** [comment]
@andre-merzky it waits for a review either from you or @lee212 . As soon as it accepted I'll merge it with devel

**Comment by andre-merzky** [comment]
Oh, apologies, will tend to it!

File: tests/test_component/test_amgr.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    #     # Global parameters to have default values
+    #     assert amgr._mqs_setup
+    #     assert amgr._autoterminate
+    #     assert not amgr._resubmit_failed
+    #     assert amgr._resource_desc is None
+    #     assert amgr._task_manager  is None
+    #     assert amgr._workflow      is None
+    #     assert amgr._reattempts  == 3
+    #     assert amgr._cur_attempt == 1
+    #     assert isinstance(amgr.shared_data, list)
```
Comment (andre-merzky) [inline tests/test_component/test_amgr.py:76]: Why is this part commented out?  Can it be removed?

**Review by andre-merzky — APPROVED** [review]
LGTM - thanks!

File: tests/test_component/test_amgr.py (hunk 1 above)
Comment (iparask) [reply tests/test_component/test_amgr.py:76]: This is the test for AMGR's constructor. I wanted to keep it as I am not sure how to go about it. I can remove it if you want

**Pull request merged**